src/cmd/mod.rs

```diff
@@ -20,57 +23,58 @@ pub enum SelectResult {
     Selected(String),
 }
 
-pub fn select_or_list_context(installed: &mut Installed) -> Result<SelectResult> {
+pub fn select_context(skim_options: &SkimOptions, installed: &mut Installed) -> Result<SelectResult> {
     installed.contexts.sort_by(|a, b| a.item.name.cmp(&b.item.name));
+    let mut context_names: Vec<_> = installed.contexts.iter().map(|c| c.item.name.clone()).collect();
 
-    if installed.contexts.is_empty() {
+    if context_names.is_empty() {
         bail!("No contexts found");
     }
 
-    if installed.contexts.len() == 1 {
-        return Ok(SelectResult::Selected(installed.contexts[0].item.name.clone()));
-    }
-
-    // We only select the context with fzf if stdout is a terminal and if
-    // fzf is present on the machine.
-    Ok(if atty::is(atty::Stream::Stdout) && fzf::is_available() {
-        match fzf::select(installed.contexts.iter().map(|c| &c.item.name))? {
-            Some(context_name) => SelectResult::Selected(context_name),
-            None => {
-                println!("Selection cancelled.");
-                SelectResult::Cancelled
-            }
+    if atty::is(atty::Stream::Stdout) {
+        // NOTE: skim show the list of context names in reverse order
+        context_names.reverse();
+        let item_reader = SkimItemReader::default();
+        let items = item_reader.of_bufread(Cursor::new(context_names.join("\n")));
+        let selected_items = Skim::run_with(skim_options, Some(items))
+            .map(|out| out.selected_items)
+            .unwrap_or_else(|| Vec::new());
+        if selected_items.is_empty() {
+            return Ok(SelectResult::Cancelled);
         }
+        Ok(SelectResult::Selected(selected_items[0].output().to_string()))
     } else {
-        for c in &installed.contexts {
-            println!("{}", c.item.name);
+        for c in context_names {
+            println!("{}", c);
         }
-        SelectResult::Listed
-    })
+        Ok(SelectResult::Listed)
+    }
 }
 
-pub fn select_or_list_namespace() -> Result<SelectResult> {
+pub fn select_namespace(skim_options: &SkimOptions) -> Result<SelectResult> {
```

Comment:
The function was called `select_or_list_namespace` because it just lists the items when the output is not a terminal. Can you restore that?

Reply:
[9adc2d0](https://github.com/sbstp/kubie/pull/117/commits/9adc2d047db23a80a8ed9fa4e8b35c7795c70abd)